packages/material-ui-lab/src/SpeedDialAction/SpeedDialAction.js

```diff
@@ -36,14 +36,32 @@ class SpeedDialAction extends React.Component {
     tooltipOpen: false,
   };
 
+  static getDerivedStateFromProps = (props, state) => {
+    if (!props.open && state.tooltipOpen) {
+      state.tooltipOpen = false;
+    }
+    return state;
+  };
+
   handleTooltipClose = () => {
+    if (this.props.tooltipOpen) return;
     this.setState({ tooltipOpen: false });
   };
 
   handleTooltipOpen = () => {
+    if (this.props.tooltipOpen) return;
     this.setState({ tooltipOpen: true });
   };
 
+  componentDidUpdate = prevProps => {
+    if (!this.props.tooltipOpen || prevProps.open === this.props.open) return;
+    if (!this.state.tooltipOpen) {
+      this.timeout = setTimeout(() => this.setState({ tooltipOpen: true }), this.props.delay + 100);
+    }
+  };
+
+  componentWillUnmount = () => this.timeout && clearTimeout(this.timeout);
```

Comment:
Just call `clearTimeout`. It won't complain about non-ids but will cause issues in browsers that actually use 0 as ids.